src/puppetlabs/puppetdb/command.clj

```diff
@@ -586,7 +586,8 @@
         not-exception? #(not (instance? Throwable %))
         attempt-exec (fn [db status]
                        (try
-                         (attempt-exec-command cmd db status response-chan stats
+                         ;; handle callback below to avoid races during cmd broadcast
+                         (attempt-exec-command (assoc cmd :callback identity) db status response-chan stats
```

Comment:
Maybe, `(constantly true)` instead of `identity`, I didn't see any places where we store the result of the callback, so it probably doesn't matter what we return from callback, but if we do it could end up storing a duplicate of the cmd, which has a user-defined size.